app/scripts/migrations/128.ts

```diff
@@ -0,0 +1,54 @@
+import { hasProperty, isObject } from '@metamask/utils';
+import { cloneDeep } from 'lodash';
+
+export const version = 128;
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+/**
+ * Migrates MATIC ticker in Network Configuration to POL ticker as per the direction in https://polygon.technology/blog/save-the-date-matic-pol-migration-coming-september-4th-everything-you-need-to-know
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, unknown>) {
```

Comment:
as you discovered in the mobile migration, we may need to update providerConfig.ticker iff polygon is the current network.  Because this will be a fix for 12.1.x where we still have the old network controller.  You can test the migration against that branch https://github.com/MetaMask/metamask-extension/tree/Version-v12.1.0

I think we'd still merge it to develop first, then cherry pick to a 12.1.x branch.  We may also tweak the migration number to slot into that release branch more conveniently.

Reply:
Updated migration to handle legacy `providerConfig` as well as a couple other cleanup things.

If we do cherry pick this from develop onto `12.1.1` I might need some hand holding to make sure I do this correctly. Not super familiar with this cherry-picking process.